samples/dnn/text_detection.cpp

```diff
@@ -0,0 +1,169 @@
+    if (argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    float confThreshold = parser.get<float>("thr");
+    float nmsThreshold = parser.get<float>("nms");
+    int inpWidth = parser.get<int>("width");
+    int inpHeight = parser.get<int>("height");
+    CV_Assert(parser.has("model"));
+    String model = parser.get<String>("model");
+
+    // Register a custom layer.
+    CV_DNN_REGISTER_LAYER_CLASS(ResizeBilinear, ResizeBilinearLayer);
+
+    // Load network.
+    Net net = readNet(model);
+
+    // Open a video file or an image file or a camera stream.
+    VideoCapture cap;
+    if (parser.has("input"))
+        cap.open(parser.get<String>("input"));
+    else
+        cap.open(0);
+
+    static const std::string kWinName = "EAST: An Efficient and Accurate Scene Text Detector";
+    namedWindow(kWinName, WINDOW_NORMAL);
+
+    std::vector<Mat> outs;
+    std::vector<String> outNames(2);
+    outNames[0] = "feature_fusion/Conv_7/Sigmoid";
+    outNames[1] = "feature_fusion/concat_3";
+
+    Mat frame, blob;
+    while (waitKey(1) < 0)
+    {
+        cap >> frame;
+        if (frame.empty())
+        {
+            waitKey();
+            break;
+        }
+
+        blobFromImage(frame, blob, 1.0, Size(inpWidth, inpHeight), Scalar(123.68, 116.78, 103.94), true, false);
+        net.setInput(blob);
+        net.forward(outs, outNames);
+
+        Mat scores = outs[0];
+        Mat geometry = outs[1];
+
+        // Decode predicted bounding boxes.
+        std::vector<RotatedRect> boxes;
+        std::vector<float> confidences;
+        decode(scores, geometry, confThreshold, boxes, confidences);
+
+        // Apply non-maximum suppression procedure.
+        std::vector<int> indices;
+        NMSBoxes(boxes, confidences, confThreshold, nmsThreshold, indices);
+
+        // Render detections.
+        Point2f ratio((float)frame.cols / inpWidth, (float)frame.rows / inpHeight);
+        for (size_t i = 0; i < indices.size(); ++i)
+        {
+            RotatedRect& box = boxes[indices[i]];
+
+            Point2f vertices[4];
+            box.points(vertices);
+            for (int j = 0; j < 4; ++j)
+            {
+                vertices[j].x *= ratio.x;
+                vertices[j].y *= ratio.y;
+            }
+            for (int j = 0; j < 4; ++j)
+                line(frame, vertices[j], vertices[(j + 1) % 4], Scalar(0, 255, 0), 1);
+        }
+
+        // Put efficiency information.
+        std::vector<double> layersTimes;
+        double freq = getTickFrequency() / 1000;
+        double t = net.getPerfProfile(layersTimes) / freq;
+        std::string label = format("Inference time: %.2f ms", t);
+        putText(frame, label, Point(0, 15), FONT_HERSHEY_SIMPLEX, 0.5, Scalar(0, 255, 0));
+
+        imshow(kWinName, frame);
+    }
+    return 0;
+}
+
+void decode(const Mat& scores, const Mat& geometry, float scoreThresh,
```

Comment:
probably, such a function should be put to objdetect module (or dnn module).

Reply:
@vpisarev. The thing is that this decoding procedure is too specific for this network. In example, x4 final feature maps reduction comparing to input images:

```cpp
// Decode a prediction.
// Multiple by 4 because feature maps are 4 time less than input image.
float offsetX = x * 4.0f, offsetY = y * 4.0f;
...
```